docs/environment_creation_from_scratch.md

```diff
@@ -1,54 +1,112 @@
+In the [meta/main.tf](../terraform/meta/main.tf) file, add the ARN of the `task-execution` role from the newly created 
+environment to the ECR module's `allow_access_by_roles` parameter. 
+
+`cd` into the meta folder and run `terraform apply` to update the meta environment with this change.
 
 ### Run a full apply
 
-Once these have been done, run a complete apply
+Now run a full apply using the command below (without targeting any modules or setting the `initial_create` var, and while
+`cd`d into the environment to be created folder).
+
+```terraform apply```
+
+If there are any issues creating the infra, it may be worth retrying the command a short while later in case AWS / Terraform
+was unable to configure a component or resource quickly enough before moving onto the next. Otherwise, you will need to 
+investigate the cause of the error.
+
+### Set up the deployment
+
+In the Core App codebase [here](https://github.com/communitiesuk/submit-social-housing-lettings-and-sales-data), you
+will need create a new pipeline for the environment (you can re-use the jobs and configuration in existing pipelines where
+appropriate).
+
+For the `aws_deploy` workflow, ensure you pass in the new account id, resource prefix and environment name.
+
+Once setup, run the deployment pipeline in order to update the ECS task definitions to run the right image from the ECR 
+(the pipeline may fail on the db migration stage depending on the state of your DB, see the next step for info).
 
 ### Set up the database
 
-If not restoring the database from a backup, or migrating it from elsewhere, run the db:setup rake task (use the ad_hoc task definition to spin it up from the app image with a command override).
+If you're not restoring the database from a backup / migrating it from elsewhere, and the db migration stage failed
+above, run a `db:setup` rake task (using the `ad_hoc` task definition to spin it up from the app image, with a command 
+override).
 
-Alternatively you probably need to run an app deployment to sort out using the right image, which will include a db migration - you can then just run db:seed as a separate task.
+Alternatively, if the db migration stage ran successfully, you can run a `db:seed` rake task separately (also using the 
+`ad_hoc` task definition), to populate the DB with some initial data.
 
-### Set up the deployment
+### Run a complete deployment
 
-In the Core App codebase [here](https://github.com/communitiesuk/submit-social-housing-lettings-and-sales-data), you will need create a new pipeline for the environment (you can re-use jobs and configuration in existing pipelines where appropriate)
-For the `aws_deploy` workflow, ensure you pass in the new account id, resource prefix and environment name
\ No newline at end of file
+Finally, run the deployment pipeline completely to bring the infrastructure and application to a complete and ready-to-use
```

Comment:
How does this differ from running the deployment as mentioned in the `### Set up the deployment` section above?

Reply:
You need to run the deployment first to update the task definitions to pull the correct container image i.e. pull an image with a name/tag that exists (rather than `latest` which doesn't). However if there is no DB setup, the deployment will fill on the migrations step (which is OK). 

You then have to go and make the DB (which you can only do if the task definitions are up to date, so you can run an ad-hoc task to do this which will pull the right image 😅). Then you'll be able to re-run the deployment from start to end successfully. 

Hope that makes sense 😀

Reply:
I've updated some of the instruction wording to make the above clearer 😃

Reply:
Nice one, thanks!